mirrord-agent/src/namespace.rs

```diff
@@ -0,0 +1,36 @@
+use std::{
+    fs::File,
+    os::unix::prelude::{IntoRawFd, RawFd},
+};
+
+use anyhow::Result;
+use async_trait::async_trait;
+use nix::sched::setns;
+use serde::{Deserialize, Serialize};
+
+#[async_trait]
+pub trait Namespace: Send + Sync {
+    async fn get_namespace(&self) -> Result<String>;
+    fn set_namespace(&self, ns_path: String) -> Result<()> {
```

Comment:
why would we put set_namespace in a trait?

Reply:
set_namespace is the same across all the types, thats why it is put in the trait. didnt want to repeat code

Reply:
it doesn't use self -> it can be a regular function.